packages/screen-orientation/android/src/main/java/io/capawesome/capacitorjs/plugins/screenorientation/ScreenOrientation.java

```diff
@@ -44,27 +44,10 @@ public ScreenOrientationChangeListener getOrientationChangeListener() {
         return orientationChangeListener;
     }
 
-    public void lock(String orientationType) {
-        switch (orientationType) {
-            case ScreenOrientationType.LANDSCAPE:
-                bridge.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE);
-                break;
-            case ScreenOrientationType.LANDSCAPE_PRIMARY:
-                bridge.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
-                break;
-            case ScreenOrientationType.LANDSCAPE_SECONDARY:
-                bridge.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE);
-                break;
-            case ScreenOrientationType.PORTRAIT:
-                bridge.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_SENSOR_PORTRAIT);
-                break;
-            case ScreenOrientationType.PORTRAIT_PRIMARY:
-                bridge.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
-                break;
-            case ScreenOrientationType.PORTRAIT_SECONDARY:
-                bridge.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_REVERSE_PORTRAIT);
-                break;
-        }
+    public void lock(@Nullable String orientationType) throws Exception {
+        final String key = (orientationType != null) ? orientationType : getCurrentOrientationType();
```

Comment:
```suggestion
        final String key = (orientationType == null) ? getCurrentOrientationType() : orientationType;
```

I find it much easier to read when it's the other way around.